### PR TITLE
feat: receiver code and tests for change admin in token factory

### PR DESCRIPTION
**Pull request by p0mvn**
Closes: #XXX

## What is the purpose of the change

Tokenfactory receiver code for change admin and tests

## Testing and Verifying

This change added tests and can be verified as follows:

`go test -timeout 30s -run ^TestChangeAdmin$ github.com/osmosis-labs/osmosis/v7/app/wasm/test`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? yes
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: app/wasm/test/messages_test.go [hunk 1]
```diff
@@ -64,6 +64,111 @@ func TestCreateDenom(t *testing.T) {
...
+		},
+		"actor is a different address": {
+			changeAdmin: &wasmbindings.ChangeAdmin{
+				Denom:           fmt.Sprintf("factory/%s/%s", actor.String(), validDenom),
+				NewAdminAddress: RandomBech32AccountAddress(),
+			},
+			actor:     RandomAccountAddress(),
+			expErrMsg: "failed changing admin from message: unauthorized account",
+		},
+		"change to the same address": { // TODO: make sure that success is expected here
```
Comment (p0mvn) [inline app/wasm/test/messages_test.go:133]: @ValarDragon should I be able to change admin to myself? Is this expected?

**Review by mattverse — APPROVED** [review]
Nice! LGTM

File: app/wasm/message_plugin.go [hunk 2]
```diff
@@ -122,6 +132,36 @@ func PerformMint(f *tokenfactorykeeper.Keeper, b *bankkeeper.BaseKeeper, ctx sdk
 	return nil
 }
 
+func (m *CustomMessenger) changeAdmin(ctx sdk.Context, contractAddr sdk.AccAddress, changeAdmin *wasmbindings.ChangeAdmin) ([]sdk.Event, [][]byte, error) {
+	err := ChangeAdmin(m.tokenFactory, ctx, contractAddr, changeAdmin)
+	if err != nil {
+		return nil, nil, sdkerrors.Wrap(err, "change admin")
```
Comment (alexanderbez) [inline app/wasm/message_plugin.go:138]:
```suggestion
		return nil, nil, sdkerrors.Wrap(err, "failed to change admin")
```

File: app/wasm/message_plugin.go [hunk 3]
```diff
@@ -122,6 +132,36 @@ func PerformMint(f *tokenfactorykeeper.Keeper, b *bankkeeper.BaseKeeper, ctx sdk
...
+	}
+	return nil, nil, nil
+}
+
+func ChangeAdmin(f *tokenfactorykeeper.Keeper, ctx sdk.Context, contractAddr sdk.AccAddress, changeAdmin *wasmbindings.ChangeAdmin) error {
+	if changeAdmin == nil {
+		return wasmvmtypes.InvalidRequest{Err: "changeAdmin was nil"}
```
Comment (alexanderbez) [inline app/wasm/message_plugin.go:145]:
```suggestion
		return wasmvmtypes.InvalidRequest{Err: "changeAdmin is nil"}
```

File: app/wasm/test/messages_test.go [hunk 4]
```diff
@@ -64,6 +64,110 @@ func TestCreateDenom(t *testing.T) {
 
 }
 
+func TestChangeAdmin(t *testing.T) {
+	const validDenom = "validdenom"
+
+	actor := RandomAccountAddress()
```
Comment (ValarDragon) [inline app/wasm/test/messages_test.go:70]: Lets rename this to token creator?

So then we distinguish whose executing the change admin function from the creator more clearly?

Comment (p0mvn) [reply app/wasm/test/messages_test.go]: Renamed

**Comment by p0mvn** [comment]
All comments are addressed. This should be ready for merge, unless there is anything else

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**